Sources/Navigation/Navigation.swift

```diff
@@ -196,23 +191,23 @@ extension Navigation {
         }
     }
 
-    mutating func didFail(_ navigation: WKNavigation? = nil, with error: WKError) {
+    func didFail(_ navigation: WKNavigation? = nil, with error: WKError) {
         assert(state != .expected, "non-started navigations should‘t receive didFail")
         if let navigation {
             self.identity.resolve(with: navigation)
         }
         self.state = .failed(error)
     }
 
-    mutating func didReceiveServerRedirect(for navigation: WKNavigation?) {
-        assert(state == .expected || state == .started, "didReceiveServerRedirect should happen after decidePolicyForNavigationAction")
+    func didReceiveServerRedirect(for navigation: WKNavigation?) {
+        assert(state == .redirected, "didReceiveServerRedirect should happen after decidePolicyForNavigationAction")
```

Comment:
This assertion may be too narrow – it fails for me when I’m trying to visit http://onet.pl (or just typing _onet.pl_ in the address bar). It works fine when I type `https://www.onet.pl` (cause there’s no redirect). Strangely enough I wasn’t able to reproduce it on any other website so far.

Reply:
Yes, I‘ve caught it yesterday while adding tests for redirects. Also I decided to reapproach how we handle redirects for Navigations to make it more usable.
This is fixed now